app/controllers/application_controller.rb

```diff
@@ -1,10 +1,83 @@
-require './config/environment'
 
+require 'pry'
+require './config/environment'
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+		set :session_secret, "password_security"
+  end
+
+  get "/" do
+		erb :index
+	end
+
+	get "/signup" do
+    if logged_in?
+      redirect to "/tweets"
+		else
+      erb :signup
+    end
+	end
+
+  get "/login" do
```

Comment:
You can probably move all these methods pertaining to signing in to a separate sessions controller.